From d44b2768f42b99c9b18cfc2766e687221cedc403 Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Wed, 9 Jan 2019 13:44:29 +0200
Subject: [PATCH 0891/1051] net: pp2: fixed wrongly configured gop2 RGMII
 configuration

Fixed rgmii port gop3 configuration of register GENCONF_CTRL0
(GENCONF_CTRL0_PORT1_RGMII)
Caused a non wotrking ethernet port, due to errornous configuration.
gop2 configuration included a gop3 related configuration.
the solution was to remove the redundant configuration.

Change-Id: I4fe0f80e3295a5bc9794a226f7aab05c978c5c29
Signed-off-by: Alex Leibovich <alexl@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/62537
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1916
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 5ba65e643ba7..2b45addc231c 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -923,7 +923,7 @@ static void mvpp22_gop_init_rgmii(struct mvpp2_port *port)
 
 	regmap_read(priv->sysctrl_base, GENCONF_CTRL0, &val);
 	if (port->gop_id == 2)
-		val |= GENCONF_CTRL0_PORT0_RGMII | GENCONF_CTRL0_PORT1_RGMII;
+		val |= GENCONF_CTRL0_PORT0_RGMII;
 	else if (port->gop_id == 3)
 		val |= GENCONF_CTRL0_PORT1_RGMII_MII;
 	regmap_write(priv->sysctrl_base, GENCONF_CTRL0, val);
-- 
2.17.1

